File: security_cam/lib/detection/face/mock_face_engine.dart
import '../../core/models.dart';
import 'face_engine.dart';

/// Test/dry-run engine: returns whatever [faces] was pre-loaded with.
class MockFaceEngine implements FaceEngine {
  final List<FaceDetection> faces = [];

  @override
  Future<void> init() async {}

  @override
  Future<List<FaceDetection>> detectFaces(ColorBitmap frame) async => List.of(faces);

  @override
  Future<void> dispose() async {}
}

File: android/app/src/test/kotlin/io/securitycam/level2/detection/face/MockFaceEngine.kt
package io.securitycam.level2.detection.face

import io.securitycam.level2.detection.ColorBitmap

/** Test/dry-run engine: returns whatever [faces] was pre-loaded with. */
class MockFaceEngine : FaceEngine {
    val faces = mutableListOf<FaceDetection>()

    override suspend fun init() {}

    override suspend fun detectFaces(frame: ColorBitmap): List<FaceDetection> = faces.toList()

    override suspend fun dispose() {}
}
